regression-test/suites/prepared_stmt_p0/prepared_stmt.groovy

```diff
@@ -0,0 +1,166 @@
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import java.math.BigDecimal;
+
+suite("test_prepared_stmt") {
+    def tableName = "tbl_prepared_stmt"
+    def user = context.config.jdbcUser
+    def password = context.config.jdbcPassword
+    def url = context.config.jdbcUrl + "&useServerPrepStmts=true"
+    // def url = context.config.jdbcUrl
+    def result1 = connect(user=user, password=password, url=url) {
+    def insert_prepared = { stmt, k1 , k2, k3, k4, k5, k6, k7, k8, k9 ->
+        java.text.SimpleDateFormat formater = new java.text.SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+        if (k1 == null) {
+            stmt.setNull(1, java.sql.Types.INTEGER);
+        } else {
+            stmt.setInt(1, k1)
+        }
+        if (k2 == null) {
+            stmt.setNull(2, java.sql.Types.DECIMAL);
+        } else {
+            stmt.setBigDecimal(2, k2)
+        }
+        if (k3 == null) {
+            stmt.setNull(3, java.sql.Types.VARCHAR);    
+        } else {
+            stmt.setString(3, k3)
+        }
+        if (k4 == null) {
+            stmt.setNull(4, java.sql.Types.VARCHAR);     
+        } else {
+            stmt.setString(4, k4)
+        }
+        if (k5 == null) {
+            stmt.setNull(5, java.sql.Types.DATE);     
+        } else {
+            stmt.setDate(5, java.sql.Date.valueOf(k5))
+        }
+        if (k6 == null) {
+            stmt.setNull(6, java.sql.Types.TIMESTAMP);     
+        } else {
+            stmt.setTimestamp(6, new java.sql.Timestamp(formater.parse(k6).getTime()))
+        }
+        if (k7 == null) {
+            stmt.setNull(7, java.sql.Types.FLOAT);
+        } else {
+            stmt.setFloat(7, k7)
+        }
+        if (k8 == null) {
+            stmt.setNull(8, java.sql.Types.DATE);
+        } else {
+            stmt.setTimestamp(8, new java.sql.Timestamp(formater.parse(k8).getTime()))
+        }
+        if (k9 == null) {
+            stmt.setNull(9, java.sql.Types.VARCHAR);
+        } else {
+            stmt.setString(9, k9)
+        }
+        exec stmt
+    }
+    sql """DROP TABLE IF EXISTS ${tableName} """
+    sql """
+        CREATE TABLE IF NOT EXISTS ${tableName} (
+                `k1` int(11) NULL COMMENT "",
+                `k2` decimalv3(27, 9) NULL COMMENT "",
+                `k3` varchar(30) NULL COMMENT "",
+                `k4` varchar(30) NULL COMMENT "",
+                `k5` date NULL COMMENT "",
+                `k6` datetime NULL COMMENT "",
+                `k7` float NULL COMMENT "",
+                `k8` datev2 NULL COMMENT "",
+                `k9` array<datetime> NULL COMMENT ""
+              ) ENGINE=OLAP
+              DUPLICATE KEY(`k1`, `k2`, `k3`)
+              DISTRIBUTED BY HASH(`k1`, k2, k3) BUCKETS 1
```

Comment:
BUCKETS 1 is too special, it's better to set n > 1

Reply:
test protocol is enough